src/libraries/System.Text.Json/src/System/Text/Json/Serialization/ReadStackFrame.cs

```diff
@@ -18,6 +18,7 @@ internal struct ReadStackFrame
         // Support JSON Path on exceptions.
         public byte[]? JsonPropertyName; // This is Utf8 since we don't want to convert to string until an exception is thown.
         public string? JsonPropertyNameAsString; // This is used for dictionary keys and re-entry cases that specify a property name.
+        internal byte[]? DictionaryKeyName; // This will contain the Utf8 Json property name that represents a dictionary key; used to defer parsing on async/re-entry.
```

Comment:
I realized that I can re-use `JsonPropertyName` for this purpose, however, `JsonPropertyName` will no longer be exclusive for exceptions.
@steveharter @layomia